datafusion/sql/src/planner.rs

```diff
@@ -622,21 +622,37 @@ pub fn object_name_to_table_reference(
     idents_to_table_reference(idents, enable_normalization)
 }
 
-struct IdentTaker(Vec<Ident>);
+struct IdentTaker {
+    normalizer: IdentNormalizer,
```

Comment:
the point of the PR is to move normalizer into `IdentTaker` so it isn't created again on each row